### PR TITLE
Updated scheduler name for multi-scheduler.

### LINKED ISSUE #41859: Change "scheduler name" field name from schedulername to schedulerName to adhere to API guidelines
Assigned to @k82cn 
Blocking issue for 1.6 release
ref/ #38871
cc/ @smarterclayton

### PR DESCRIPTION
**Pull request by k82cn**
fixes #41859

### CODE REVIEW

**Comment by k82cn** [comment]
The name in `pkg` was updated; I'm going to update staging part by `./hack/staging-client-go.sh` and update vender accordingly. @caesarxuchao , suppose the above steps is correct :).

cc @smarterclayton , @davidopp for status update.

```
staging/src/k8s.io/client-go/pkg/api/v1/generated.proto:2621:  optional string schedulername = 19;
staging/src/k8s.io/client-go/pkg/api/v1/types.generated.go:31261:					r.EncodeString(codecSelferC_UTF81234, string("schedulername"))
staging/src/k8s.io/client-go/pkg/api/v1/types.generated.go:31556:		case "schedulername":
staging/src/k8s.io/client-go/pkg/api/v1/types_swagger_doc_generated.go:1312:	"schedulername":                 "If specified, the pod will be dispatched by specified scheduler. If not specified, the pod will be dispatched by default scheduler.",
vendor/k8s.io/client-go/pkg/api/v1/generated.proto:2621:  optional string schedulername = 19;
vendor/k8s.io/client-go/pkg/api/v1/types.generated.go:31261:					r.EncodeString(codecSelferC_UTF81234, string("schedulername"))
vendor/k8s.io/client-go/pkg/api/v1/types.generated.go:31556:		case "schedulername":
vendor/k8s.io/client-go/pkg/api/v1/types_swagger_doc_generated.go:1312:	"schedulername":                 "If specified, the pod will be dispatched by specified scheduler. If not specified, the pod will be dispatched by default scheduler.",
```

**Comment by caesarxuchao** [comment]
@k82cn `hack/staging-client-go.sh` is correct. Please let me know if the script doesn't work.

**Comment by smarterclayton** [comment]
Please bump client-go in a separate commit for audit and reviewing purposes.

**Comment by k82cn** [comment]
@caesarxuchao , it seems `hack/staging-client-go.sh` did not update example of client-go, which file should I copy manually?

```
staging/src/k8s.io/apiserver/pkg/apis/example/v1/generated.proto:173:  optional string schedulername = 19;
staging/src/k8s.io/apiserver/pkg/apis/example/v1/types.go:178:	SchedulerName string `json:"schedulername,omitempty" protobuf:"bytes,19,opt,name=schedulername"`
vendor/k8s.io/apiserver/pkg/apis/example/v1/generated.proto:173:  optional string schedulername = 19;
vendor/k8s.io/apiserver/pkg/apis/example/v1/types.go:178:	SchedulerName string `json:"schedulername,omitempty" protobuf:"bytes,19,opt,name=schedulername"`
```

**Comment by k82cn** [comment]
@k8s-bot gci gke e2e test this

**Comment by k82cn** [comment]
@k8s-bot unit test this

**Comment by k82cn** [comment]
@k8s-bot verify test this

**Comment by k82cn** [comment]
@k8s-bot unit test this

**Comment by k82cn** [comment]
@k8s-bot non-cri e2e test this

**Comment by k82cn** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by k82cn** [comment]
@lavalamp , and suggestion on how to update the following files after running `./hack/update-staging-client.sh` ?

```
staging/src/k8s.io/apiserver/pkg/apis/example/v1/generated.proto:173:  optional string schedulername = 19;
staging/src/k8s.io/apiserver/pkg/apis/example/v1/types.go:178:	SchedulerName string `json:"schedulername,omitempty" protobuf:"bytes,19,opt,name=schedulername"`
```

**Comment by caesarxuchao** [comment]
`staging/src/k8s.io/apiserver/pkg/apis/example/v1/` is authoritative, so are the examples in client-go. You can modify them directly. Sorry it's confusing.

**Comment by k82cn** [comment]
@caesarxuchao , as we slacked , the original version of those files are generated, so I'd like to which one are they :).

@deads2k , any suggestion on the original version of `staging/src/k8s.io/apiserver/pkg/apis/example/v1/generated.proto` and `staging/src/k8s.io/apiserver/pkg/apis/example/v1/types.go`? I'd like to update them manually :).

**Comment by k82cn** [comment]
@k8s-bot kops aws e2e test this

**Comment by k82cn** [comment]
@k8s-bot bazel test this

**Comment by k82cn** [comment]
/cc @davidopp , @smarterclayton 

Slack @deads2k , `staging/src/k8s.io/apiserver/pkg/apis/example` is used for unit test; I'd suggest to merge this PR firstly, I'll open another PR for `example`.

**Comment by deads2k** [comment]
> Slack @deads2k , staging/src/k8s.io/apiserver/pkg/apis/example is used for unit test; I'd suggest to merge this PR firstly, I'll open another PR for example.

Yeah, the copy used in a unit test isn't a blocker and word is it wasn't trivial to update there because something went weird with generation.

**Review by timothysc — APPROVED** [review]
You need to rebase and possibly regen again.

**Comment by k82cn** [comment]
@timothysc , sure; will rebase & regen soon.

**Comment by k82cn** [comment]
#42031 will trace to sync `staging/src/k8s.io/apiserver/pkg/apis/example`.

**Comment by k82cn** [comment]
@davidopp , @smarterclayton ; "Updated scheduler name for multi-scheduler." PR is ready for review :).

**Comment by timothysc** [comment]
This change is so small, green lighting... b/c I saw the previous version was the same.

**Comment by k82cn** [comment]
@timothysc , as a `release-broker`, can we increase it priority?

**Comment by davidopp** [comment]
It's not necessary to increase the priority. Because this is a bug fix, it's fine if it goes in after the code freeze.

**Comment by k82cn** [comment]
Great :).

**Comment by k82cn** [comment]
failed to rebase because of https://github.com/kubernetes/kubernetes/issues/42133

**Comment by k82cn** [comment]
@timothysc , rebased :).

**Comment by davidopp** [comment]
There are changes in this PR that seem to be unrelated to changing schedulername -> schedulerName. Could you please figure out how to reduce the PR to only that change?

**Comment by timothysc** [comment]
@davidopp This is the result of changing apis.

**Comment by davidopp** [comment]
I think we should have someone from @kubernetes/sig-api-machinery-misc review this PR. I am not qualified to know whether this PR is right.

**Comment by liggitt** [comment]
the first two commits look right (but need a rebase). I did not expect the client-go commit

**Comment by k82cn** [comment]
@liggitt , the conflict (staging) is updated by `hack/update-staging-client-go.sh` which is client-go commit.

cc @caesarxuchao for client-go part.

**Comment by liggitt** [comment]
Hmm. @sttts @caesarxuchao is that script required to be run now? I think @sttts just merged a PR that would contain most of those changes already

**Comment by caesarxuchao** [comment]
Can you remove your third commit, rebase, and then run hack/update-staging-client.sh again? As liggitt suggested, sttts's recent PR should have checked in most of the changes.

**Comment by k82cn** [comment]
sure, I will update it today :).

**Comment by k82cn** [comment]
Rebased. less modified files now :).

File: staging/src/k8s.io/client-go/Godeps/Godeps.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
 	"ImportPath": "k8s.io/client-go",
-	"GoVersion": "go1.7",
-	"GodepVersion": "v74",
+	"GoVersion": "go1.8",
+	"GodepVersion": "v79",
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/Godeps/Godeps.json]: revert this change

Comment (timothysc) [inline staging/src/k8s.io/client-go/Godeps/Godeps.json]: Should not be updating godeps at all.  

How are you generating?  You'd have be calling update-godeps.?.?.?

Comment (liggitt) [inline staging/src/k8s.io/client-go/Godeps/Godeps.json]: update-staging-client-go does godeppy things, but it looks like it tries to force a specific godep version. @caesarxuchao, any tips here?

Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/Godeps/Godeps.json]: update-staging-client-go.sh forces godep v74, see this [line](https://github.com/kubernetes/kubernetes/blob/master/hack/update-staging-client-go.sh#L25). I suspect staging/copy.sh was run directly, however, that should give you a [warning](https://github.com/kubernetes/kubernetes/blob/master/staging/copy.sh#L49).

Comment (k82cn) [inline staging/src/k8s.io/client-go/Godeps/Godeps.json]: I run `update-staging-client-go.sh` in Mac; godep are download manually.

Comment (k82cn) [inline staging/src/k8s.io/client-go/Godeps/Godeps.json]: BTW, how about go version? should I change it back to 1.7?

Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/Godeps/Godeps.json]: Yes, please.

**Comment by k82cn** [comment]
@k8s-bot bazel test this

**Comment by k82cn** [comment]
rebased :).

**Comment by k82cn** [comment]
rebased :).

**Comment by k82cn** [comment]
need rebase again :(. I'll do that.

**Comment by timothysc** [comment]
@k82cn  FAILED   hack/make-rules/../../hack/verify-generated-protobuf.sh	89s

**Comment by k82cn** [comment]
will update today.

**Comment by k82cn** [comment]
rebased :).

**Comment by k82cn** [comment]
@k8s-bot cvm gke e2e test this

**Comment by k82cn** [comment]
@k8s-bot cvm gke e2e test this

**Comment by k82cn** [comment]
@k8s-bot gci gke e2e test this

**Comment by calebamiles** [comment]
@smarterclayton, @davidopp could you please add some more context here as to why this PR is a release blocker? Thanks!

**cc: @ethernetdan, @kubernetes/kubernetes-release-managers**

**Comment by liggitt** [comment]
The original PR typoed the API field. If we release that, we'll be locked into backwards compatibility support of an API with a typo

**Comment by k82cn** [comment]
@k8s-bot cvm gke e2e test this

**Comment by k82cn** [comment]
@k8s-bot cvm gke e2e test this
@k8s-bot gci gke e2e test this

**Comment by k82cn** [comment]
ping @timothysc for lgtm label :)

**Comment by k82cn** [comment]
@k8s-bot bazel test this

**Comment by k82cn** [comment]
@k8s-bot gci gke e2e test this
@k8s-bot kops aws e2e test this
@k8s-bot kubemark e2e test this
@k8s-bot cvm gke e2e test this

**Comment by k82cn** [comment]
@k8s-bot gci gke e2e test this

**Comment by k82cn** [comment]
@k8s-bot gci gke e2e test this
